### PR TITLE
Refactor tx_pool

### PR DESCRIPTION
**Pull request by zp-sd**
Simplify `tx_pool`, `txs`, `top` and `tx_pool_feeder`:
* `lists:reverse` instead of `flip`
* better naming & formatting

### CODE REVIEW

File: apps/ae_core/src/consensus/chain/top.erl [hunk 1]
```diff
@@ -1,57 +1,94 @@
...
+                if
+                    I == M -> keys:update_id(1);
+                    true -> ok
+                end,
+                block:hash(GenesisMakerBlock);
+            Top ->
+                Top
+        end,
+    spawn(fun() ->
+                  block_hashes:add(block:hash(GenesisMakerBlock))
```
Comment (sennui) [inline apps/ae_core/src/consensus/chain/top.erl]: can we compute just once `block:hash(GenesisMakerBlock)` and pass it to block_hashes AND state of the top gen_server AND block_internal ?  cc: @zack-bitcoin

File: apps/ae_core/src/consensus/chain/top.erl [hunk 2]
```diff
@@ -1,57 +1,94 @@
...
+handle_cast(_Msg, State) ->
+    {noreply, State}.
+
+handle_info(_Info, State) ->
+    {noreply, State}.
 
-test() ->
-    ok.
+terminate(_Reason, _State) ->
+    ok = lager:warning("Top died!").
```
Comment (sennui) [inline apps/ae_core/src/consensus/chain/top.erl]: when you are around can you add `?MODULE` in printout

File: apps/ae_core/src/consensus/tx_pool.erl [hunk 3]
```diff
@@ -1,50 +1,93 @@
...
+terminate(_Reason, _State) ->
+    ok = lager:warning("Tx pool died!").
+
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+
+%% Internals
+
+current_state() ->
```
Comment (sennui) [inline apps/ae_core/src/consensus/tx_pool.erl]: can we keep one in state? so we dont read from db? cc: @zack-bitcoin

File: apps/ae_core/src/consensus/tx_pool_feeder.erl [hunk 4]
```diff
@@ -1,61 +1,94 @@
...
+    {ok, State}.
 
-is_in(_, []) -> false;
-is_in(STx, [STx2|T]) ->
+
+%% Internals
+
+is_in(_, []) ->
+    false;
+is_in(STx, [STx2 | T]) ->
```
Comment (sennui) [inline apps/ae_core/src/consensus/tx_pool_feeder.erl]: can we keep pre signed transactions in state? List is not the best for read perf also

**Comment by sennui** [comment]
@zp-sd we need to chat about it with Zack to propose potential improvements. I think some stuff can be pre-computed in state (digest?)

File: apps/ae_core/src/consensus/chain/top.erl (hunk 2 above)
Comment (zp-sd) [inline apps/ae_core/src/consensus/chain/top.erl]: _changing_ 👍

File: apps/ae_core/src/consensus/chain/top.erl (hunk 1 above)
Comment (zp-sd) [inline apps/ae_core/src/consensus/chain/top.erl]: This makes sense, let's wait on @zack-bitcoin and refactor in next step of refactoring, wdyt?

File: apps/ae_core/src/consensus/tx_pool.erl (hunk 3 above)
Comment (zp-sd) [inline apps/ae_core/src/consensus/tx_pool.erl]: Sounds feasible, can we wait for next phase of refactoring?

**Comment by zp-sd** [comment]
@sennui 💯 agree, however I'd start with this PR as cleanup, and apply more improvements in next batch.

File: apps/ae_core/src/consensus/tx_pool.erl (hunk 3 above)
Comment (zp-sd) [inline apps/ae_core/src/consensus/tx_pool.erl]: @sennui Hm, `top:doit()` returns the most recent hash block stored in `top` gen_server (it is not always the same), I'm not sure if we can keep it in state.

**Pull request merged**